app/src/main/java/me/hufman/androidautoidrive/phoneui/viewmodels/MusicActivityModel.kt

```diff
@@ -12,18 +12,20 @@ import me.hufman.androidautoidrive.music.MusicAppInfo
 import me.hufman.androidautoidrive.music.MusicController
 import me.hufman.androidautoidrive.music.QueueMetadata
 import me.hufman.androidautoidrive.music.controllers.SpotifyAppController
+import me.hufman.androidautoidrive.music.spotify.SpotifyWebApi
 
-class MusicActivityModel(val musicController: MusicController): ViewModel() {
+class MusicActivityModel(val musicController: MusicController, val spotifyWebApi: SpotifyWebApi): ViewModel() {
 	class Factory(val appContext: Context, val musicApp: MusicAppInfo): ViewModelProvider.Factory {
 		@Suppress("UNCHECKED_CAST")
 		override fun <T : ViewModel?> create(modelClass: Class<T>): T {
 			var model: MusicActivityModel? = null
 			val controller = MusicController(appContext, Handler())
+			val spotifyWebApi = SpotifyWebApi.getInstance(appContext)
```

Comment:
I don't think `MusicActivityModel`'s constructor should know about the SpotifyWebApi. I think the leaking of Spotify details should remain in `updateErrors`, and that function should dig into the SpotifyAppController for web api authentication errors. SpotifyAppController might need to have a surface function to get the web api authentication status, to reduce the depth of the needed mocks a tiny bit.

Reply:
This one is complicated since the dependency injection pattern would have it so that we should have the SpotifyWebApi passed in the constructor in order to decouple these classes a little bit. Also while we can do the SpotifyWebApi.getInstance(..) in the updateErrors method, we run into some slight problems. First of which is that we don't have a clean way to access the SpotifyAppController to access the SpotifyWebApi and any of it's methods from the MusicController instance I don't believe. While this can be worked around by just using the SpotifyWebApi.getInstance in the updateErrors method, this again seems like a dependency injection avoidance and also has the not fun side effect of making the code harder to test. Do we want to go with this approach and avoid the SpotifyWebApi instance in the constructor?

Also on kind of a similar note I feel like the getAuthorized() for the SpotifyWebApi shouldn't be a part of the SpotifyAppController as that class is not responsible for the SpotifyWebApi internal state and it seems a little out of place